Sourcery/Parsing/FileParser.swift

```diff
@@ -344,20 +344,21 @@ extension FileParser {
     fileprivate func parseEnumAssociatedValues(_ body: String) -> [Enum.Case.AssociatedValue] {
         guard !body.isEmpty else { return [] }
 
-        /// name: type, otherType
-        let components = body.components(separatedBy: ",")
-        return components.enumerated().flatMap { idx, element in
-            let nameType = element.components(separatedBy: ":").map { $0.trimmingCharacters(in: .whitespacesAndNewlines) }
-
-            switch nameType.count {
-            case 1:
-                return Enum.Case.AssociatedValue(name: "\(idx)", typeName: nameType.first ?? "")
-            case 2:
-                return Enum.Case.AssociatedValue(name: nameType.first, typeName: nameType.last ?? "")
-            default:
-                print("\(logPrefix)parseEnumAssociatedValues: Unknown enum case body format \(body)")
-                return nil
-            }
+        let items = body.commaSeparated()
+        return items
+            .map({ $0.trimmingCharacters(in: .whitespaces) })
+            .enumerated()
+            .map {
+                let nameAndType = $1.colonSeparated().map({ $0.trimmingCharacters(in: .whitespaces) })
+                let defaultName: String? = $0 == 0 && items.count == 1 ? nil : "\($0)"
```

Comment:
@krzysztofzablocki any ideas what names better to use for that? `localName` and `externalName`?

To better illustrate the issue here are few cases:

```swift
enum Some {
  case option0(Int)
  case option1(v: Int)
  case option2(v: Int, Int)
}

let opt1 = Some.option0(1) //no local name
let opt2 = Some.option1(v: 1) //has local name
let opt3 = Some.option2(v: 1, 2) //first has local name, second has no local name

switch some {
  case let .option0(a): print(a) // has no external name
  case let .option1(a): print(a) // has no external name
  case let .option2(a): print(a.first); print(a.1) //first has external name, second has index name
}
```